src/Pages/RegisterForm/RegisterForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import { Button, Form, Input } from 'antd';
+import type { FormItemProps } from 'antd';
+import './style.scss';
+
+const MyFormItemContext = React.createContext<(string | number)[]>([]);
+
+interface MyFormItemGroupProps {
+  prefix: string | number | (string | number)[];
+}
+
+function toArr(
+  str: string | number | (string | number)[]
+): (string | number)[] {
+  return Array.isArray(str) ? str : [str];
+}
+
+const MyFormItemGroup: React.FC<
+  React.PropsWithChildren<MyFormItemGroupProps>
+> = ({ prefix, children }) => {
+  const prefixPath = React.useContext(MyFormItemContext);
+  const concatPath = React.useMemo(
+    () => [...prefixPath, ...toArr(prefix)],
+    [prefixPath, prefix]
+  );
+
+  return (
+    <MyFormItemContext.Provider value={concatPath}>
+      {children}
+    </MyFormItemContext.Provider>
+  );
+};
+
+const MyFormItem = ({ name, ...props }: FormItemProps) => {
+  const prefixPath = React.useContext(MyFormItemContext);
+  const concatName =
+    name !== undefined ? [...prefixPath, ...toArr(name)] : undefined;
+
+  return <Form.Item name={concatName} {...props} />;
+};
+
+const App: React.FC = () => {
+  const [form] = Form.useForm();
+
+  const [formData, setFormData] = useState({
+    firstName: '',
+    businessEmail: '',
+    companyName: '',
+  });
+  const [errorName, setErrorName] = useState<string | null>(null);
+  const [errorEmail, setErrorEmail] = useState<string | null>(null);
+  const [errorCompany, setErrorCompany] = useState<string | null>(null);
+
+  const validateForm = () => {
+    let valid = true;
+
+ 
+    if (!/^[\p{L}\d\s._-]+$/u.test(formData.firstName)) {
+      setErrorName('Please enter only letters and the field cannot be empty');
+      valid = false;
+    } else {
+      setErrorName(null);
+    }
+
+
+    if (
+      !/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/.test(
+        formData.businessEmail
+      )
+    ) {
+      setErrorEmail('Enter correct e-mail');
```

Comment:
antd provided great validation of form fields, their inbuilt logic  btw. (https://2x.ant.design/components/form/#components-form-demo-validate-static)  But this is cool to practice of course

Reply:
Used validation from ant Design